### PR TITLE
 script.sh: gnupg

### PR DESCRIPTION
**Pull request by commitay**
https://github.com/Homebrew/brew/pull/4120

The travis pre-installed `gnupg` is relatively recent but is missing several required dependancies (no idea if they would impact the functionality we need), seems easier just to upgrade it to the current version and install the missing deps which would also mirror the setup that most users have.
```
>>> brew info gnupg
gnupg: stable 2.2.6 (bottled)
GNU Pretty Good Privacy (PGP) package
https://gnupg.org/
/usr/local/Cellar/gnupg/2.2.1 (132 files, 10.3MB)
  Poured from bottle on 2017-09-19 at 20:01:30
/usr/local/Cellar/gnupg/2.2.3 (133 files, 10.3MB) *
  Poured from bottle on 2017-12-06 at 04:46:19
From: https://github.com/Homebrew/homebrew-core/blob/master/Formula/gnupg.rb
==> Dependencies
Build: pkg-config ✔
Required: npth ✔, gnutls ✘, libgpg-error ✘, libgcrypt ✘, libksba ✔, libassuan ✘, pinentry ✘, gettext ✔, adns ✔
```

### CODE REVIEW

File: ci/travis/script.sh [hunk 1]
```diff
@@ -30,18 +30,22 @@ for file in "${modified_ruby_files[@]}"; do
...
   run brew cask style "${modified_casks[@]}"
   if [[ ${#modified_casks[@]} -le 3 ]]; then
     if /usr/bin/grep "depends_on cask:" "${modified_casks[@]}" > /dev/null; then
-      export HOMEBREW_NO_AUTO_UPDATE=1
       run brew tap homebrew/bundle
       run brew bundle dump --file="${HOME}/Brewfile"
     fi
     for cask in "${modified_casks[@]}"; do
+      if brew cask _stanza gpg "${cask}" > /dev/null; then
+        run brew outdated gnupg || run brew upgrade gnupg
```
Comment (vitorgalvao) [inline ci/travis/script.sh]: If we’re setting `HOMEBREW_NO_AUTO_UPDATE` earlier, then won’t `brew outdated` not properly check for the actual latest version?

Comment (commitay) [reply ci/travis/script.sh]: Actually, this won't work because the audit step will also check the gpg signature.

**Comment by commitay** [comment]
> Yes, this should be checked before `_audit_modified_casks`.

Done.

Not sure what's causing this:
```
==> Failed command:
/usr/local/opt/gnupg/bin/gpg --no-default-keyring --keyring /Users/travis/Library/Caches/Homebrew/gpg/homebrew-keyring.gpg --receive-keys 2007B954
==> Standard Output of failed command:
==> Standard Error of failed command:
gpg: keybox '/Users/travis/Library/Caches/Homebrew/gpg/homebrew-keyring.gpg' created
gpg: keyserver receive failed: No route to host
```

**Comment by reitermarkus** [comment]
> Not sure what's causing this:

Probably just a temporary network issue.

**Comment by commitay** [comment]
This seems to be working well with travis now, I've tried several Casks.

**Pull request merged**